src/transactions.ts

```diff
@@ -379,9 +391,11 @@ function makeScanResultFromScanIteratorInternal<
 async function* getScanIteratorForIndexMap(
   dbRead: db.Read,
   options: ScanIndexOptions,
-) {
+): AsyncIterable<IndexKeyEntry> {
   const map = await dbRead.getMapForIndex(options.indexName);
-  yield* map.scan(fromKeyForIndexScanInternal(options));
+  for await (const entry of map.scan(fromKeyForIndexScanInternal(options))) {
```

Comment:
And here is where we convert the internal iterator from encoded strings to IndexKey